lib/ansible/modules/system/portenforce.py

```diff
@@ -0,0 +1,192 @@
+version_added: "2.3"
+description:
+    - Kill processes listening on TCP and UDP ports not defined in the whitelists.
+short_description: Kill processes listening on TCP and UDP ports not defined in the whitelists.
+options:
+  whitelist_tcp:
+    description:
+      - A list of TCP ports to allow.
+    required: false
+    type: "list"
+  whitelist_udp:
+    description:
+      - A list of UDP ports to allow.
+    required: false
+    type: "list"
+'''
+
+EXAMPLES = '''
+- name: only allow TCP 80, 443, and 22
+  portenforce:
+    whitelist_tcp:
+      - 22
+      - 80
+      - 443
+'''
+
+RETURN = '''
+killed:
+  description: A list of pids that were killed.
+  type: list
+  contains:
+    pid:
+      description: The pid of a killed process.
+      type: int
+      sample: 1223
+    port:
+      description: The port the killed pid was listening on.
+      type: int
+      sample: 443
+    proto:
+      description: The protocol, TCP or UDP, of the listening connection.
+      type: string
+      sample: 'tcp'
+    stime:
+      description: The start time of the killed process.
+      type: string
+      sample: Thu Apr 21 17:30:16 2016
+'''
+
+import re
+import os
+import platform
+from subprocess import Popen, PIPE
+
+def netStatParse(raw):
+    results = list()
+    for line in iter(raw.splitlines()):
+        listening_search = re.search('[^ ]+:[^ ]+', line)
+        if listening_search:
+            splitted = line.split()
+            conns = re.search('[^ ]+:([^ ]+)', splitted[3])
+            pidstr = ''
+            if 'tcp' in splitted[0]:
+                proto = 'tcp'
+                pidstr = splitted[6]
+            elif 'udp' in splitted[0]:
+                proto = 'udp'
+                pidstr = splitted[5]
+            pids = re.search('([0-9]+)/(.*)', pidstr)
+            if conns and pids:
+                port = conns.group(1)
+                pid = pids.group(1)
+                name = pids.group(2)
+                result = dict(pid=int(pid), port=int(port), proto=proto, name=name)
+                if result not in results:
+                    results.append(result)
+            elif not pids:
+                raise EnvironmentError('Could not get the pids for the listening ports - possibly a permission issue')
+    return results
+
+def applyWhitelist(portspids, whitelist=list()):
+    kill_pids = list()
+    for p in portspids:
+        if int(p['port']) not in whitelist and str(p['port']) not in whitelist:
+            to_kill = dict(pid=p['pid'], port=p['port'], proto=p['proto'], name=p['name'], stime=p['stime'])
+            if to_kill not in kill_pids:
+                kill_pids.append(to_kill)
+    return kill_pids
+
+def main():
+
+    module = AnsibleModule(
+        argument_spec = dict(
+            whitelist_tcp = dict(required=False, type='list', default=list()),
+            whitelist_udp = dict(required=False, type='list', default=list())
+        ),
+        supports_check_mode=True
+    )
+
```

Comment:
Thanks, yes this would be ideal - I've added a check in https://github.com/ansible/ansible/pull/19475/commits/1d5ff4f2f53aa9cb2c0a303ae0a7daed01c03108 that hits a `module.fail_json()` call if both are missing.